File: leetcode/easy/sum-of-digits-of-string-after-convert.py
# https://leetcode.com/problems/sum-of-digits-of-string-after-convert/

class Solution:
    def getLucky(self, s: str, k: int) -> int:
        s = "".join(str(ord(ch)-96) for ch in s)
        for _ in range(k):
            s = str(sum(int(ch) for ch in s))
        return int(s)


# The ord() function returns an integer representing the Unicode character.

File: leetcode/easy/sum-of-digits-of-string-after-convert.ts
// https://leetcode.com/problems/sum-of-digits-of-string-after-convert/

// MINE
function getLucky(s: string, k: number): number {
  // convert
  let str = ''
  for (let char of s) str += `${char.charCodeAt(0) - 96}`

  // transform
  for (let i = 0; i < k; ++i) {
    let temp = 0
    for (let digit of str) temp += parseInt(digit)
    str = `${temp}`
  }

  return parseInt(str)
}

// ANSWER
